core/templates/pages/skill-editor-page/skill-editor-page.component.ts

```diff
@@ -89,23 +91,10 @@ angular.module('oppia').component('skillEditorPage', {
         return ctrl.skill.getValidationIssues().length;
       };
 
-      ctrl.setUpBeforeUnload = function() {
-        WindowRef.nativeWindow.addEventListener(
-          'beforeunload', ctrl.confirmBeforeLeaving);
-      };
-
-      ctrl.confirmBeforeLeaving = function(e) {
-        if (UndoRedoService.getChangeCount()) {
-          // This message is irrelevant, but is needed to trigger the
-          // confirmation before leaving.
-          e.returnValue = 'Sure?';
-          return false;
-        }
-      };
-
       ctrl.$onInit = function() {
         BottomNavbarStatusService.markBottomNavbarStatus(true);
-        ctrl.setUpBeforeUnload();
+        PreventPageUnloadEventService.addListener(
+          () => UndoRedoService.getChangeCount());
```

Comment:
Here and other places:
Can we do `PreventPageUnloadEventService.addListener(UndoRedoService.getChangeCount)`?

Reply:
PTAL https://github.com/oppia/oppia/pull/12742/files#r640067259

Reply:
DONE